Woof/Woof/ViewModel/SitterProfileViewModel.swift

```diff
@@ -30,20 +37,32 @@ final class SitterProfileViewModel: ObservableObject {
         setInitialValues(currentSitter)
     }
 
-    /// Requests model layer to save modified data.
-    func save() {
-        currentSitter.name = name
-        currentSitter.surname = surname
-        currentSitter.phone = phone
-        currentSitter.bio = bio
-        if let newPricePerHour = Double(pricePerHour) {
-            currentSitter.pricePerHour = newPricePerHour
+    /// Requests model layer to upload and save modified data.
+    func save() async {
+        await MainActor.run {
+            isSavingData = true
         }
 
-        guard let data = try? JSONEncoder().encode(currentSitter) else { return }
+        var updatedSitter = currentSitter
+
+        updatedSitter.name = name
+        updatedSitter.surname = surname
+        updatedSitter.phone = phone
+        updatedSitter.bio = bio
+        updatedSitter.pricePerHour = Double(pricePerHour) ?? currentSitter.pricePerHour
+
+        do {
+            try await upload(updatedSitter)
+
+            currentSitter = updatedSitter
 
-        KeyValueStorage(KeyValueStorage.Name.currentSitter)
-            .save(data, for: KeyValueStorage.Key.currentSitter)
+            await saveLocally(currentSitter)
+        } catch {
+            await handleError(.uploadFailed)
+        }
```

Comment:
We have 2 errors of `AppError` type, but here you expect only one type. Why?
Or we don't need 2 cases of AppError.

Reply:
The second error type could trigger in `saveLocally()`